api/src/main/java/no/bibsys/db/RegistryManager.java

```diff
@@ -15,115 +14,99 @@
 
 public class RegistryManager {
 
-    private final transient TableDriver tableDriver;
+    private final transient TableDriver tableManager;
+    private final transient ItemDriver itemManager;
+    private final transient ObjectMapper objectMapper = ObjectMapperHelper.getObjectMapper();
 
-    public RegistryManager(TableDriver tableDriver) {
-        this.tableDriver = tableDriver;
+    public RegistryManager(TableDriver tableManager, ItemDriver itemManager) {
+        this.tableManager = tableManager;
+        this.itemManager = itemManager;
     }
-    
-    public void createRegistry(EntityRegistryTemplate request)
-            throws InterruptedException, TableAlreadyExistsException, JsonProcessingException {
-        TableManager tableManager = new TableManager(tableDriver);
-
-        String tableName = request.getId();
-        if (registryExists(tableName)) {
-            throw new TableAlreadyExistsException(String.format("Registry %s already exists", tableName));
-        } else {
-            tableManager.createRegistry(request);
-        }
-    }
-    
-    public void addRegistry(String tableName, String json) throws JsonProcessingException, InterruptedException {
-        if (!registryExists(tableName)) {
-            EntityManager tableWriter = new EntityManager(tableDriver, tableName);
-            tableWriter.addJson(json);
-        } else {
-            throw new TableAlreadyExistsException(String.format("Registry %s already exists", tableName));
-        }
-        
+
+    public void createRegistry(EntityRegistryTemplate request) throws JsonProcessingException {
+        String registryName = request.getId();
+        String json = objectMapper.writeValueAsString(request);
+        addRegistry(registryName, json);
     }
-    
-    public boolean registryExists(String tableName) throws InterruptedException {
-        return new TableManager(tableDriver).registryExists(tableName);
+
+    public void addRegistry(String registryName, String json) throws JsonProcessingException{
+
+        tableManager.createTable(getValidationSchemaTable());
+
+        itemManager.addItem(getValidationSchemaTable(), json);
+        tableManager.createTable(registryName);
+
     }
 
-    public void emptyRegistry(String tableName) throws InterruptedException {
-        TableManager tableManager = new TableManager(tableDriver);
-        tableManager.emptyTable(tableName);
+    public boolean registryExists(String tableName) {
+        return tableManager.tableExists(tableName);
+    }
 
+    public void emptyRegistry(String tableName) {
+        tableManager.deleteTable(tableName);

```

Comment:
Should be implemented as a function in tableDriver

Reply:
Registry is not a table, it's a table + item in the registry schema table